Source/SwiftLintFramework/Rules/QuickDiscouragedCallRule.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  QuickDiscouragedCallRule.swift
+//  SwiftLint
+//
+//  Created by Ornithologist Coder on 8/11/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+// swiftlint:disable identifier_name
+
+public struct QuickDiscouragedCallRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "quick_discouraged_call",
+        name: "Quick Discouraged Call",
+        description: "Discouraged call inside 'describe' and/or 'context' block.",
+        kind: .style,
+        nonTriggeringExamples: QuickDiscouragedCallRuleExamples.nonTriggeringExamples,
+        triggeringExamples: QuickDiscouragedCallRuleExamples.triggeringExamples
+    )
+
+    public func validate(file: File,
+                         kind: SwiftExpressionKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard
+            fileContainsQuickSpec(file: file),
```

Comment:
I've made a minor change to address the first comment. Now it checks if the call is within the bounds of the QuickSpec class.

For instance:

```swift
import Foundation
import Quick

public class FooBar {
    func foo() {
        describe("too") {
            let bar = Bar()
        }
    }

    func bar() {
        context("too") {
            let bar = Bar()
        }
    }
}

private class TotoTests: QuickSpec {
    override func spec() {
        describe("top level describe") {
            let foo = Foo() // violation expected
            let bar = Bar() // violation expected

            context("first level context 1") {
                let foo = Foo() // violation expected

                context("second level context") {
                    beforeEach {
                        let foo = Foo()
                        foo.toto()
                    }

                    it("test") {
                        let foo = Foo()
                        foo.toto()
                    }

                    context("third level context") {
                        let foo = Foo() // violation expected
                    }
                }
            }

            context("first level context 2") {
                let foo = Foo() // violation expected
            }

            context("first level context 3") {
                foo.toto() // violation expected
            }

            context("first level context 4") {
                var foo: Foo!

                beforeEach {
                    foo = Foo()
                }

                it("foo") {  }
            }

            context("first level context 5") {
                var foo: Foo!

                beforeSuite {
                    foo = Foo()
                }

                it("foo") {  }
            }
        }
    }
}

internal class BarFoo {
    func foo() {
        describe("too") {
            let bar = Bar()
        }
    }

    func bar() {
        context("too") {
            let bar = Bar()
        }
    }
}
```

returns these violations:

```
$ ./.build/debug/swiftlint lint --path ~/Desktop/quick.swift --enable-all-rules --no-cache
Loading configuration from '.swiftlint.yml'
Linting Swift files at path /Users/ornithocoder/Desktop/quick.swift
Linting 'quick.swift' (1/1)
/Users/ornithocoder/Desktop/quick.swift:1: warning: File Header Violation: Header comments should be consistent with project patterns. (file_header)
/Users/ornithocoder/Desktop/quick.swift:39:35: warning: Quick Discouraged Call Violation: Discouraged call inside a 'context' block. (quick_discouraged_call)
/Users/ornithocoder/Desktop/quick.swift:25:27: warning: Quick Discouraged Call Violation: Discouraged call inside a 'context' block. (quick_discouraged_call)
/Users/ornithocoder/Desktop/quick.swift:45:27: warning: Quick Discouraged Call Violation: Discouraged call inside a 'context' block. (quick_discouraged_call)
/Users/ornithocoder/Desktop/quick.swift:49:17: warning: Quick Discouraged Call Violation: Discouraged call inside a 'context' block. (quick_discouraged_call)
/Users/ornithocoder/Desktop/quick.swift:21:23: warning: Quick Discouraged Call Violation: Discouraged call inside a 'describe' block. (quick_discouraged_call)
/Users/ornithocoder/Desktop/quick.swift:22:23: warning: Quick Discouraged Call Violation: Discouraged call inside a 'describe' block. (quick_discouraged_call)
Done linting! Found 7 violations, 0 serious in 1 file.
```